sbi/simulators/simutils.py

```diff
@@ -117,56 +117,59 @@ def simulate_in_batches(
     """
-    Draw `num_samples` parameter sets and simulate them in batches of size `simulation_batch_size`.
+    Draw `num_samples` parameter sets and simulate them in batches of size
+     `simulation_batch_size`.
 
     Features: - allows to simulate in batches of arbitrary size
               - if `simulation_batch_size==-1`, all simulations are run at the same time
               - simulator output can be np.array or torch.Tensor
 
     Args:
-        simulator: simulator function.
-            If `simulation_batch_size == 1`: takes in parameters of shape (1, num_dim_parameters)
-                and outputs xs of shape (1, num_dim_x)
-            If `simulation_batch_size > 1`: takes in thetas of shape (simulation_batch_size, num_dim_parameters)
-                and outputs xs of shape (simulation_batch_size, num_dim_x)
+        simulator: simulator function that takes in parameters theta and outputs data x
+            If `simulation_batch_size == 1`: takes in parameters of shape
+             (1, num_dim_parameters) and outputs xs of shape (1, num_dim_x)
+            If `simulation_batch_size > 1`: takes in thetas of shape
+             (simulation_batch_size, num_dim_parameters) and outputs xs of shape
+             (simulation_batch_size, num_dim_x)
         parameter_sample_fn: Function to call for generating theta, e.g. prior sampling
         num_samples: Number of simulations to run
         simulation_batch_size: Number of simulations that are run within a single batch
-            If `simulation_batch_size == -1`, we run a batch with all simulations required,
-            i.e. `simulation_batch_size = num_samples`
-        x_dim: dimensionality of a single simulator output
+            If `simulation_batch_size == -1`, we run a batch with all simulations
+             required, i.e. `simulation_batch_size = num_samples`
 
-    Returns: torch.Tensor simulation input parameters of shape (num_samples, num_dim_parameters),
+    Returns: torch.Tensor simulation input parameters of shape
+                (num_samples, num_dim_parameters)
              torch.Tensor simulator outputs x of shape (num_samples, num_dim_x)
     """
 
-    # generate parameters (simulation inputs) by sampling from prior (round 1) or proposal (round > 1)
+    # generate parameters (simulation inputs) by sampling from prior (round 1)
+    # or proposal (round > 1)
     parameters = parameter_sample_fn(num_samples)
 
     if simulation_batch_size == -1:
         # run all simulations in a single batch
         simulation_batch_size = num_samples
 
-    # split parameter set into batches of size (simulation_batch_size, num_dim_parameters)
+    # split parameter set into batches of size
+    # (simulation_batch_size, num_dim_parameters)
     n_chunks = math.ceil(num_samples / simulation_batch_size)
     parameter_batches = torch.chunk(parameters, chunks=n_chunks)
 
     all_x = []
     for batch in parameter_batches:
         with torch.no_grad():
```

Comment:
Not part of this PR, but I believe  this context manager should be out of the loop?